nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/main/java/org/apache/nifi/parameter/aws/AwsSecretsManagerParameterProvider.java

```diff
@@ -70,16 +70,38 @@
 @CapabilityDescription("Fetches parameters from AWS SecretsManager.  Each secret becomes a Parameter group, which can map to a Parameter Context, with " +
         "key/value pairs in the secret mapping to Parameters in the group.")
 public class AwsSecretsManagerParameterProvider extends AbstractParameterProvider implements VerifiableParameterProvider {
+    public static final String ENUMERATED_STRATEGY = "Enumerated";
+    public static final String PATTERN_STRATEGY = "Pattern";
+    public static final PropertyDescriptor SECRET_LISTING_STRATEGY = new PropertyDescriptor.Builder()
+            .name("secret-listing-strategy")
+            .displayName("Secret Listing Strategy")
+            .description("Strategy to use when listing secrets. 'Pattern' strategy uses Secret Name Pattern property and fetches  " +
+                    "all secrets whose names match the pattern. 'Pattern' strategy requires ListSecrets and GetSecretValue permissions. "+
+                    "'Enumerated' strategy uses 'Secret Names' property and fetches all secrets in the list. " +
+                    "'Enumerated' strategy requires only GetSecretValue permission.")
+            .required(true)
+            .allowableValues(PATTERN_STRATEGY, ENUMERATED_STRATEGY)
```

Comment:
In case you're opting for using an enum, this can be simplified to:

```suggestion
            .allowableValues(ListingStrategy.class)
```